sway-core/src/type_system/type_parameter.rs

```diff
@@ -220,8 +220,25 @@ impl TypeParameter {
                     type_arguments: trait_type_arguments,
                 } = trait_constraint;
 
+                // Use trait name with module path as this is expected in get_methods_for_type_and_trait_name
+                let mut full_trait_name = trait_name.clone();
+                if trait_name.prefixes.is_empty() {
+                    if let Some(vec) = ctx.namespace.use_synonyms.get(&trait_name.suffix) {
+                        full_trait_name = CallPath {
```

Comment:
Why only the first element from vec?

Reply:
The vec variable is in fact `(Vec<Ident>, GlobImport)`. I will change its name to cause less confusion.